Insight.Database/Reliable/RetryStrategy.cs

```diff
@@ -95,21 +95,38 @@ public TResult ExecuteWithRetry<TResult>(IDbCommand commandContext, Func<TResult
 			{
 				try
 				{
-					return func();
+                    return func();
 				}
 				catch (Exception ex)
 				{
 					// if it's not a transient error, then let it go
-					if (!IsTransientException(ex))
-						throw;
+				    if (!IsTransientException(ex))
+				    {
+				        commandContext.AutoClose();

```

Comment:
Don't close on a normal throw.
Insight's normal behavior is that if you give it an open connection, it should stay open when you return it.